core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java

```diff
@@ -245,6 +248,36 @@ public static DataType getDataType(String dataTypeStr) {
   }
 
   /**
+   * This method will convert the data according to its data type and perform a
+   * special handling for decimal data types
+   *
+   * @param dataInBytes
+   * @param dimension
+   * @return
+   */
+  public static Object getDataBasedOnDataType(byte[] dataInBytes, CarbonDimension dimension) {
```

Comment:
change the previous function to handle the logic
 getDataBasedOnDataType(dataInBytes, dimension.getDataType());